contrib/google.golang.org/grpc/option.go

```diff
@@ -1,14 +1,25 @@
 package grpc
 
+import (
+	"google.golang.org/grpc/codes"
+)
+
 // Option specifies a configuration option for the grpc package. Not all options apply
 // to all instrumented structures.
 type Option = InterceptorOption
 
 type config struct {
-	serviceName                           string
-	traceStreamCalls, traceStreamMessages bool
-	noDebugStack                          bool
-	analyticsRate                         float64
+	serviceName string
+
+	// codes.Code values will be used to index into this slice.
+	// Given a codes.Code "c" the expression nonErrorCodes[c]
+	// yields whether this code is not to be regarded as an error.
+	nonErrorCodes []bool
```

Comment:
BTW I've decided to not use a hashmap here, since the assigned `codes.Code` integers can be represented as a small, dense 17-byte boolean array.
Just in case I wrote `setNonErrorCodes`/`isNotAnErrorCode` in such a way that it can deal with unknown status codes though.